client/state/jetpack-agency-dashboard/reducer.ts

```diff
@@ -0,0 +1,20 @@
+import { LICENSE_PURCHASED_VIA_AGENCY_DASHBOARD } from './action-types';
+import type { PurchasedProduct } from 'calypso/jetpack-cloud/sections/agency-dashboard/sites-overview/types';
+
+interface LicensePurchasedAction {
+	type: typeof LICENSE_PURCHASED_VIA_AGENCY_DASHBOARD;
+	payload: PurchasedProduct | null;
+}
+
+type AgencyDashboardReducerAction = LicensePurchasedAction;
```

Comment:
Why do we need to define an interface to define a type? I don't think we are using the interface anywhere else, aren't we?

Reply:
Action types might vary, since this is a reducer for the agency dashboard, we might add more actions in the future. So, keep it future-proof I have kept it like this.